modin/core/dataframe/pandas/partitioning/axis_partition.py

```diff
@@ -510,7 +526,18 @@ def deploy_func_between_two_axis_partitions(
         with warnings.catch_warnings():
             warnings.filterwarnings("ignore", category=FutureWarning)
             result = func(lt_frame, rt_frame, *f_args, **f_kwargs)
-        return split_result_of_axis_func_pandas(axis, num_splits, result)
+        if return_generator:
+            return generate_result_of_axis_func_pandas(
```

Comment:
This wouldnt work because using yeild in a function would turn it to a generator.

We do not require generators but lists for some branches of if , For the  backends such as dask as we try to return a list of partitions, but as there is yield statement a generator would still be returned and thus partitons would be empty when materialized.

https://stackoverflow.com/questions/26595895/return-and-yield-in-the-same-function

Reply:
Checked with `yield from` as well still the function returns a generator when called so the code fails for dask.